### ISSUE TITLE
build error: redeclaration of succcess_cb 

### ISSUE THREAD

**Issue by theoparis** [fubark/cosmic#39]
```
zig build gen -Darg="api-js" -Darg="runtime/snapshots/gen_api.js" -Dfetch -Dtarget=native-linux-musl
/theodata/projects/theos-test/cosmic/runtime/work_queue.zig:297:16: error: redeclaration of function parameter 'success_cb'
            fn success_cb(_ctx_ptr: *anyopaque, ptr: *anyopaque) void {
               ^~~~~~~~~~
/theodata/projects/theos-test/cosmic/runtime/work_queue.zig:292:18: note: previous declaration here
        comptime success_cb: fn (std.meta.Child(@TypeOf(ctx_ptr)), TaskOutput(TaskImpl)) void,
                 ^~~~~~~~~~
error: gen...
```

**Comment by theoparis** [comment]
Even more errors on the latest zig master when i removed those parameters:
![image](https://user-images.githubusercontent.com/11761863/185032216-d7167636-1bc4-44bb-8114-e055f1fa5dc7.png)

**Comment by theoparis** [comment]
Got even more errors after fixing that one.

![image](https://user-images.githubusercontent.com/11761863/185032302-07b6c943-42be-44d1-8d8b-b57fde3e3a07.png)

**Comment by fubark** [comment]
This should be fixed by #40. Even though the js runtime passes the tests, the graphics api has definitely fallen behind. If you're interested in the latest graphics/ui developments you can check out their respective subprojects for instructions on how to run the examples.
